### PR TITLE
[12.0][IMP] web_advanced_search: support hierarchy filtering

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Add `is child of` and `is parent of` operators to filtering
image

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@Tardo @Yajo @chienandalu maybe you'd be interested in this

File: web_advanced_search/static/src/js/web_advanced_search.js [hunk 1]
```diff
@@ -153,14 +156,23 @@ odoo.define("web_advanced_search", function (require) {
             // To make widgets work, we need a model and an empty record
             FieldManagerMixin.init.call(this);
             this.trigger_up("get_dataset");
-            // Make equal and not equal appear 1st and 2nd
+            // Unfortunately, it's impossible to check if model supports these
```
Comment (legalsylvain) [inline web_advanced_search/static/src/js/web_advanced_search.js:159]: Why ?  Checking if model has parent_id field is not possible ?

**Review by legalsylvain — COMMENTED** [review]
One question. LGTM otherwise.
Thanks !
(Code review / no test)

File: web_advanced_search/static/src/js/web_advanced_search.js (hunk 1 above)
Comment (alexey-pelykh) [reply web_advanced_search/static/src/js/web_advanced_search.js:159]: Technically, we should check if `_parent_store = True` or check for field with name stored in `_parent_name` (by default, that's `parent_id`), yet I haven't found any clues on how to get those in frontend or example of those values being used in js.

Comment (legalsylvain) [reply web_advanced_search/static/src/js/web_advanced_search.js:159]: OK ! Could you just add it in the roadmap section ?

Comment (alexey-pelykh) [reply web_advanced_search/static/src/js/web_advanced_search.js:159]: Done

**Review by legalsylvain — APPROVED** [review]

**Review by yajo — APPROVED** [review]

**Pull request merged**